src/cdk-experimental/menu/menu-stack.ts

```diff
@@ -55,15 +55,43 @@ export class MenuStack {
   }
 
   /**
-   *  Pop off the top most MenuStackItem and emit it on the close observable.
+   *  Pop off the top most MenuStackItem and emit it on the close observable if the stack is not
+   *  empty. If `lastElement` is provided it will pop off until this element (inclusive). If
+   * `lastElement` is  not in the stack, it will pop off the topmost element only.
+   *  @param lastElement the last item to pop off the stack.
    *  @param focusNext the event to emit on the `empty` observable if the method call resulted in an
    *  empty stack. Does not emit if the stack was initially empty.
    */
-  closeLatest(focusNext?: FocusNext) {
-    const menuStackItem = this._elements.pop();
-    if (menuStackItem) {
-      this._close.next(menuStackItem);
-      if (this._elements.length === 0) {
+  closeInclusive(lastElement?: MenuStackItem, focusNext?: FocusNext) {
+    if (!this.isEmpty()) {
+      const containsElement = lastElement ? this._elements.indexOf(lastElement) >= 0 : false;
+      let poppedElement;
+      do {
+        poppedElement = this._elements.pop();
+        this._close.next(poppedElement);
+      } while (containsElement && poppedElement !== lastElement && !this.isEmpty());
+
+      if (this.isEmpty()) {
+        this._empty.next(focusNext);
+      }
+    }
+  }
+
+  /**
+   *  Pop off the top most MenuStackItem and emit it on the close observable if the stack is not
+   *  empty. If `lastElement` is provided it will pop off until this element is the last one on the
+   *  stack. If `lastElement` is not in the stack, it will pop off the topmost element only.
+   *  @param lastElement the element which should be left on the stack
+   *  @param focusNext the event to emit on the `empty` observable if the method call resulted in an
+   *  empty stack. Does not emit if the stack was initially empty.
+   */
+  closeExclusive(lastElement?: MenuStackItem, focusNext?: FocusNext) {
```

Comment:
Hmm, now there are two methods that do the same thing if you don't provide a parameter, but behave differently if you do. Are you sure these two methods definitely need the first parameter to be optional? Also, can you achieve the effect you need with a pair of methods that don't overlap? (Eg. 'closeExclusive() and closeLatest()'.) That way, you might be able to avoid the slightly redundant method implementations.